### PR TITLE
Pack refactor downsized

### PR DESCRIPTION
**Pull request by kumaryash90**

### CODE REVIEW

**Comment by nkrishang** [comment]
When initializing `transferRole`, etc. in the `initialize` function, use memory variables in internal function calls (e.g. `_setupRole`) instead of the actual state variables (`transferRole`, etc.).

If the contract exceeds the size limit due to this, disregard this suggestion.

**Comment by nkrishang** [comment]
Check whether adding `ERC1155Receiver` is necessary in the top level `supportsInterface` function.

**Comment by nkrishang** [comment]
In `addPackContents`, add the following check: `require(balanceOf[recipient][packId] > 0)`

File: contracts/pack/Pack.sol [hunk 1]
```diff
@@ -94,9 +95,9 @@ contract Pack is
         address _royaltyRecipient,
         uint256 _royaltyBps
     ) external initializer {
-        transferRole = keccak256("TRANSFER_ROLE");
-        minterRole = keccak256("MINTER_ROLE");
-        assetRole = keccak256("ASSET_ROLE");
+        bytes32 _transferRole = keccak256("TRANSFER_ROLE");
+        bytes32 _minterRole = keccak256("MINTER_ROLE");
+        bytes32 _assetRole = keccak256("ASSET_ROLE");
```
Comment (kumaryash90) [inline contracts/pack/Pack.sol:100]: local variables when initializing roles

**Comment by kumaryash90** [comment]
applied the changes @nkrishang

**Pull request merged**